src/Service/PartyHelper.php

```diff
@@ -126,4 +126,73 @@ public function handleDeleteParty(CaseEntity $case, Party $party)
 
         $this->entityManager->flush();
     }
+
+    public function getAllPartyTypes(CaseEntity $case): array
+    {
+        $complainantPartyTypes = $this->getComplainantPartyTypesByCase($case);
+        $counterPartyTypes = $this->getCounterPartyTypesByCase($case);
+
+        return array_merge($complainantPartyTypes, $counterPartyTypes);
+    }
+
+    public function getComplainantPartyTypesByCase(CaseEntity $case): array
+    {
+        $rawTypes = explode(
+            PHP_EOL,
+            $case->getBoard()->getComplainantPartyTypes()
+        );
+
+        return $this->getTrimmedTypes($rawTypes);
+    }
+
+    public function getCounterPartyTypesByCase(CaseEntity $case): array
+    {
+        $rawTypes = explode(
+            PHP_EOL,
+            $case->getBoard()->getCounterPartyTypes()
+        );
+
+        return $this->getTrimmedTypes($rawTypes);
+    }
+
+    public function getTrimmedTypes(array $rawTypes): array
+    {
+        $trimmedTypes = [];
+        foreach ($rawTypes as $rawType) {
+            $trimmedRawType = trim($rawType);
+            $trimmedTypes[$trimmedRawType] = $trimmedRawType;
+        }
+
+        return $trimmedTypes;
+    }
+
+    /**
+     * Returns array containing relevant party arrays.
+     *
+     * @return array[]
+     */
+    public function getRelevantPartiesByCase(CaseEntity $case): array
+    {
+        $complainantRelations = $this->relationRepository
+            ->findBy([
+                'case' => $case,
+                'type' => $this->getComplainantPartyTypesByCase($case),
+                'softDeleted' => false,
+            ])
+        ;
+
+        $complainants = array_map(function ($relation) { return ['party' => $relation->getParty(), 'type' => $relation->getType()]; }, $complainantRelations);
+
+        $counterPartyRelations = $this->relationRepository
+            ->findBy([
+                'case' => $case,
+                'type' => $this->getCounterPartyTypesByCase($case),
+                'softDeleted' => false,
+            ])
+        ;
+
+        $counterparties = array_map(function ($relation) { return ['party' => $relation->getParty(), 'type' => $relation->getType()]; }, $counterPartyRelations);
```

Comment:
I think this should be formatted different. The body of the function should be on a new line etc.

Reply:
Agreed. Increased code readability.